data/src/test/java/org/apache/iceberg/TestPartitionStatsHandler.java

```diff
@@ -82,6 +65,11 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.api.io.TempDir;
 
+/**
+ * This test covers {@link PartitionStatsHandler} from the core module. Since it relies on {@link
```

Comment:
We started discussing some of this on Slack with @ajantha-bhat but for the record and more visibility:
I think it's a bit off that the Util/Handler implementations are in core module while (because of the usage of InternalData) it's not feasible to move the relevant tests to the same place and they remain in the data module. It might be the symptom of code being on the wrong place.

Would it make sense and would it be feasible to move the implementation code to the data module so that they can live at the same place as their tests?

Reply:
Will move the test to core (same module as source code), make it abstract and extend it in respective modules to supply format. 
https://github.com/apache/iceberg/pull/12060#issuecomment-2853282661

Reply:
updated.